sktime/forecasting/model_selection/_split.py

```diff
@@ -834,6 +835,9 @@ def __init__(
         self.step_length = step_length
         self.start_with_window = start_with_window
         self.initial_window = initial_window
+        if self.initial_window == 0:
+            self.start_with_window = False
```

Comment:
this block violates the `sklearn` interface conventions since it the self parameters are not equal to the init parameters.

The tests do not catch this, since `initial_window = 0` is not a test case, so please add that as a test case in `get_test_params`.